pkg/runner/runner.go

```diff
@@ -0,0 +1,224 @@
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// MarshalJSON - override the marshal json.
+func (e EventTime) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("\"%s\"", e.Time.Format("2006-01-02T15:04:05.99999999"))), nil
+}
+
+// JobEvent - event of an ansible run.
+type JobEvent struct {
+	UUID      string                 `json:"uuid"`
+	Counter   int                    `json:"counter"`
+	StdOut    string                 `json:"stdout"`
+	StartLine int                    `json:"start_line"`
+	EndLine   int                    `json:"EndLine"`
+	Event     string                 `json:"event"`
+	EventData map[string]interface{} `json:"event_data"`
+	PID       int                    `json:"pid"`
+	Created   EventTime              `json:"created"`
+}
+
+// StatusJobEvent - event of an ansible run.
+type StatusJobEvent struct {
+	UUID      string         `json:"uuid"`
+	Counter   int            `json:"counter"`
+	StdOut    string         `json:"stdout"`
+	StartLine int            `json:"start_line"`
+	EndLine   int            `json:"EndLine"`
+	Event     string         `json:"event"`
+	EventData StatsEventData `json:"event_data"`
+	PID       int            `json:"pid"`
+	Created   EventTime      `json:"created"`
+}
+
+// StatsEventData - data for a the status event.
+type StatsEventData struct {
+	Playbook     string         `json:"playbook"`
+	PlaybookUUID string         `json:"playbook_uuid"`
+	Changed      map[string]int `json:"changed"`
+	Ok           map[string]int `json:"ok"`
+	Failures     map[string]int `json:"failures"`
+	Skipped      map[string]int `json:"skipped"`
+}
+
+// Runner - a runnable that should take the parameters and name and namespace
+// and run the correct code.
+type Runner interface {
+	Run(map[string]interface{}, string, string, string) (*StatusJobEvent, error)
+}
+
+// Playbook - playbook type of runner.
+type Playbook struct {
+	Path string
+	GVK  schema.GroupVersionKind
+}
+
+// Run - This should allow the playbook runner to run.
+func (p *Playbook) Run(parameters map[string]interface{}, name, namespace, kc string) (*StatusJobEvent, error) {
+	parameters["meta"] = map[string]string{"namespace": namespace, "name": name}
+	runnerSandbox := fmt.Sprintf("/home/ansible-operator/runner/%s/%s/%s/%s/%s", p.GVK.Group, p.GVK.Version, p.GVK.Kind, namespace, name)
+	b, err := json.Marshal(parameters)
+	if err != nil {
+		return nil, err
+	}
+	//Write parameters to correct file on disk
+	err = createRunnerEnvironment(b, runnerSandbox, kc)
+	if err != nil {
+		return nil, err
+	}
+	//copy playbook from the path to the project of the runnerEnvironment
+	err = copyFile(p.Path, fmt.Sprintf("%v/project/playbook.yaml", runnerSandbox))
+	if err != nil {
+		return nil, err
+	}
+	ident := rand.Int()
+	logrus.Infof("running: %v for playbook: %v", ident, p.Path)
+
+	dc := exec.Command("ansible-runner", "-vv", "-p", "playbook.yaml", "-i", fmt.Sprintf("%v", ident), "run", runnerSandbox)
```

Comment:
Would we want the verbosity to be configurable?

Reply:
I think we need a good mechanism for this and should follow up on this.